lib/facility.rb

```diff
@@ -1,14 +1,97 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :phone,
+              :address,
+              :services,
+              :collected_fees,
+              :registered_vehicles
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_details)
+    @name = facility_details[:name]
+    @phone = facility_details[:phone]
+    @address = facility_details[:address]
     @services = []
+    @collected_fees = 0
+    @registered_vehicles = []
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if services.include?("Vehicle Registration")
+      @registered_vehicles << vehicle
+      vehicle.set_date
+      if  vehicle.antique? # If vehicle is older than 25 years, it is :antique
+        @collected_fees += 25 # Add $25 to @collected_fees
+        vehicle.set_plate(:antique)
+      elsif vehicle.electric_vehicle? # Elsif vehicle = EV
+        @collected_fees += 200    # Add $200 to @collected_fees
+        vehicle.set_plate(:ev)
+      else
+        @collected_fees += 100 # Else add $100 to @collected_fees
+        vehicle.set_plate(:regular)
+      end
+    end
```

Comment:
This method looks good! I like that you wrote a method in the Vehicle class called `set_date` to handle setting the date. 
The other thing we could do to refactor this, is take the if block from lines 26-35 and put that in it's own "helper method", maybe it could be called `set_fees_and_plate` and then the `register_vehicle` method would just call that method on line 26, but all the code/logic would be inside that helper method. That's how we would go about refactoring this.